internal/characters/arlecchino/arlecchino.go

```diff
@@ -0,0 +1,130 @@
+package arlecchino
+
+import (
+	"fmt"
+
+	tmpl "github.com/genshinsim/gcsim/internal/template/character"
+	"github.com/genshinsim/gcsim/pkg/core"
+	"github.com/genshinsim/gcsim/pkg/core/action"
+	"github.com/genshinsim/gcsim/pkg/core/attributes"
+	"github.com/genshinsim/gcsim/pkg/core/event"
+	"github.com/genshinsim/gcsim/pkg/core/glog"
+	"github.com/genshinsim/gcsim/pkg/core/info"
+	"github.com/genshinsim/gcsim/pkg/core/keys"
+	"github.com/genshinsim/gcsim/pkg/core/player/character"
+	"github.com/genshinsim/gcsim/pkg/model"
+)
+
+func init() {
+	core.RegisterCharFunc(keys.Arlecchino, NewChar)
+}
+
+type char struct {
+	*tmpl.Character
+	skillDebt             float64
+	skillDebtMax          float64
+	initialDirectiveLevel int
+	chargeEarlyCancelled  bool
```

Comment:
this variable here should be removed in favour of checking previous action + param to see if last action is an early cancelled charge attack. more robust than a bool that you have to remember to unset at the cost of some performance maybe (not enough to matter i think)